src/jit/codegenlinear.cpp

```diff
@@ -335,6 +328,25 @@ void CodeGen::genCodeForBBlist()
         // as we encounter it.
         CLANG_FORMAT_COMMENT_ANCHOR;
 
+#ifdef DEBUG
+        // Set the use-order numbers for each node.
+        {
+            unsigned useNum = 0;
+            for (GenTree* node : LIR::AsRange(block).NonPhiNodes())
+            {
+                assert((node->gtDebugFlags & GTF_DEBUG_NODE_CG_CONSUMED) == 0);
+
+                node->gtUseNum = -1;
+                for (GenTree* operand : node->Operands())
+                {
+                    assert(operand->gtUseNum == -1);
+                    operand->gtUseNum = useNum;
+                    useNum++;
```

Comment:
This seems like a reasonable approach. If one were to get fancy, one could maintain the `LsraLocation` at which it is consumed, along with its operand index - but that would be more info than is needed here (though possibly handy for debugging purposes!)